### PR TITLE
Update server.py

### PR DESCRIPTION
**Pull request by ghafers**
Changes in function: def xml_format to write dictionary to a file 
def xml_format_dict : added for dictionary console
The version is tested for XML @Hjadallah94 @gartharmstrong Please check

### CODE REVIEW

File: server/server.py [hunk 1]
```diff
@@ -1,27 +1,52 @@
...
 
-def xml_format(file_name): # function for XML(temporary) (added to code)
+
+def xml_format(file_name): # function for XML
+    with open(file_name,'r') as pkfile: # added to code
+        unpickled = et.parse(pkfile) # added to code
+        root = unpickled.getroot()
+        write_file = et.tostring(root)
+        xml_decode = write_file.decode() 
+        xml_dict = xmltodict.parse(xml_decode) #change XML to dictionary
```
Comment (gartharmstrong) [inline server/server.py]: Replace line 38: xml_dict = xmltodict.parse(xml_decode)
with
        xml_datadict = xmltodict.parse(xml_decode) #change XML to dictionary
        xml_dict = dict(xml_datadict.pop('data'))

This results in the nested ordered dictionary being retrieved and converted back to dict.

Comment (gartharmstrong) [inline server/server.py]: Replace xml_dict = xmltodict.parse(xml_decode)
with
        xml_datadict = xmltodict.parse(xml_decode) #change XML to dictionary
        xml_dict = dict(xml_datadict.pop('data'))

This results in the nested ordered dictionary being retrieved and converted back to dict.

File: server/server.py [hunk 2]
```diff
@@ -1,27 +1,52 @@
...
+
+def xml_format_dict(file_name): # function for XML <class 'dict'>
     with open(file_name,'r') as pkfile: # added to code
         unpickled = et.parse(pkfile) # added to code
-    with open(file_name, 'w') as unpkfile: # added to code
-        unpkfile.write(str(unpickled)) # added to code
+        root = unpickled.getroot()
+        write_file = et.tostring(root)
+        xml_decode = write_file.decode() # 
+        xml_dict = xmltodict.parse(xml_decode) #change XML to dictionary
```
Comment (gartharmstrong) [inline server/server.py]: Replace line 48: xml_dict = xmltodict.parse(xml_decode)
with
xml_datadict = xmltodict.parse(xml_decode) #change XML to dictionary
xml_dict = dict(xml_datadict.pop('data'))

This results in the nested ordered dictionary being retrieved and converted back to dict.

Comment (gartharmstrong) [inline server/server.py]: Replace xml_dict = xmltodict.parse(xml_decode)
with
xml_datadict = xmltodict.parse(xml_decode) #change XML to dictionary
xml_dict = dict(xml_datadict.pop('data'))

This results in the nested ordered dictionary being retrieved and converted back to dict.

**Review by gartharmstrong — CHANGES_REQUESTED** [review]
I've added 2 comments to the code. Making these changes should result in the server side dictionary being identical to the client side - maybe not in the same order but format is the same.

File: server/server.py (hunk 2 above)
Comment (gartharmstrong) [reply server/server.py]: Closing as this change is a duplicate of the above change.

**Comment by ghafers** [comment]
xml changes working fine - save some lines of code great!

File: server/server.py (hunk 1 above)
Comment (ghafers) [reply server/server.py]: Please go ahead with the changes

Comment (gartharmstrong) [reply server/server.py]: Changes have been added

File: server/server.py (hunk 2 above)
Comment (gartharmstrong) [reply server/server.py]: Changes have been added

**Review by gartharmstrong — APPROVED** [review]
Tested and ok to be merged.

**Pull request merged**